main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala

```diff
@@ -119,6 +119,8 @@ object WeededAst {
 
     case class LetRec(ident: Name.Ident, ann: Annotations, mod: Ast.Modifiers, exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
 
+    case class LocalDef(ann: Annotations, ident: Name.Ident, fparams: List[FormalParam], declaredTpe: Option[Type], declaredEff: Option[Type], exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
```

Comment:
maybe rename to `declTpe` and `declEff` instead of `declaredTpe` and `declaredEff`

Reply:
I am ok with declaredTpe and declaredEff. What is used elsewhere in this file?

Reply:
What about Jonathan's comment that it should be `expectedTpe` and `expectedEff` like `Ascribe`?